From 2f411512be54ef626536d5eb2c8b24e60e636142 Mon Sep 17 00:00:00 2001
From: Georgy Yakovlev <168902+gyakovlev@users.noreply.github.com>
Date: Tue, 21 Dec 2021 16:44:18 -0800
Subject: [PATCH] zfs-test/mmap_seek: fix build on musl

The build on musl needs linux/fs.h for SEEK_DATA and friends,
and sys/sysmacros.h for P2ROUNDUP.  Add the needed headers.

Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Georgy Yakovlev <gyakovlev@gentoo.org>
Closes #12891
---
 tests/zfs-tests/cmd/mmap_seek/mmap_seek.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/zfs-tests/cmd/mmap_seek/mmap_seek.c b/tests/zfs-tests/cmd/mmap_seek/mmap_seek.c
index f476e1dba..bb36527aa 100644
--- a/tests/zfs-tests/cmd/mmap_seek/mmap_seek.c
+++ b/tests/zfs-tests/cmd/mmap_seek/mmap_seek.c
@@ -29,7 +29,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/sysmacros.h>
 #include <errno.h>
+#ifdef __linux__
+#include <linux/fs.h>
+#endif
 
 static void
 seek_data(int fd, off_t offset, off_t expected)
-- 
2.34.1

